src/Compilers/CSharp/Portable/Binder/LockBinder.cs

```diff
@@ -54,6 +54,16 @@ internal override BoundStatement BindLockStatementParts(BindingDiagnosticBag dia
                 Error(diagnostics, ErrorCode.ERR_LockNeedsReference, exprSyntax, exprType);
                 hasErrors = true;
             }
+            else if (exprType.IsWellKnownTypeLock() &&
+                Compilation.GetWellKnownType(WellKnownType.System_Threading_Lock__Scope) is { } scopeType)
```

Comment:
>Compilation.GetWellKnownType(WellKnownType.System_Threading_Lock__Scope) is { } scopeType [](http://example.com/codeflow?start=16&length=89)

During binding we usually also collect use-site info for all additional types and members that we know lowering/codegen is absolutely going to use.